launcher/cog-launcher.c

```diff
@@ -238,11 +238,14 @@ cog_launcher_create_view(CogLauncher *self, CogShell *shell)
     }
 
     g_autoptr(GError) error = NULL;
-    CogPlatform      *platform =
-        cog_platform_configure(s_options.platform_name, s_options.platform_params, "COG", shell, &error);
+    CogPlatform      *platform = cog_platform_new(s_options.platform_name ?: g_getenv("COG_PLATFORM_NAME"), &error);
     if (!platform)
         g_error("Cannot instantiate a platform: %s", error->message);
 
+    if (!cog_platform_setup(platform, shell, s_options.platform_params ?: (g_getenv("COG_PLATFORM_PARAMS") ?: ""),
```

Comment:
... and this could be immediately after the `cog_platform_new` inside of the  `if (!platform)`

Reply:
I mean, the:

``` c
if (!platform)
        g_error("Cannot instantiate a platform: %s", error->message);
```

could be inside of 

```
    CogPlatform      *platform = cog_platform_get_default();
	    if (!platform) {
	        platform = cog_platform_new(s_options.platform_name ?: g_getenv("COG_PLATFORM_NAME"), &error);
	    }   
```

Reply:
As agreed in a chat with @psaavedra, changed the patch to move things a bit more around and have a `cog_init()` function + `cog_platform_get()`.